File: 2017/fall/8/weeks/1/positive.py
# Abstraction and scope

from cs50 import get_int


def main():
    i = get_positive_int("positive integer, please: ")
    print(f"{i} is a positive integer")


def get_positive_int(prompt):
    """Prompt user for positive integer"""
    while True:
        n = get_int(prompt)
        if n > 0:
            break
    return n


if __name__ == "__main__":
    main()

File: 1/src1/positive.c
// Abstraction and scope

#include <cs50.h>
#include <stdio.h>

int get_positive_int(string prompt);

int main(void)
{
    int i = get_positive_int("positive integer, please: ");
    printf("%i\n", i);
}

// Prompt user for positive integer
int get_positive_int(string prompt)
{
    int n;
    do
    {
        n = get_int(prompt);
    }
    while (n < 1);
    return n;
}
